app/controllers/assessments_controller.rb

```diff
@@ -181,34 +181,46 @@ def update
 
     assessment = Assessment.find_by(id: params.permit(:id)[:id])
     reported_symptoms_array = params.permit({ symptoms: %i[name value type label notes required] }).to_h['symptoms']
+    redirect_to(root_url) && return if reported_symptoms_array.any? { |symptom| symptom[:name].nil? }
 
-    typed_reported_symptoms = Condition.build_symptoms(reported_symptoms_array)
+    valid_symptom_names = assessment.reported_condition&.threshold_condition&.symptoms&.pluck(:name)
+    redirect_to(root_url) && return if valid_symptom_names.nil? || (reported_symptoms_array.map { |symptom| symptom[:name] } - valid_symptom_names).any?
+
+    reported_symptoms = reported_symptoms_array.map { |symptom| [symptom[:name], symptom] }.to_h
 
-    # Figure out the change
     delta = []
-    typed_reported_symptoms.each do |symptom|
-      new_val = symptom.value
-      old_val = assessment.reported_condition&.symptoms&.find_by(name: symptom.name)&.value
-      case symptom.type
-      when 'BoolSymptom'
-        has_changed = old_val != new_val && !old_val.nil? && !new_val.nil?
-        delta << "#{symptom.label} (\"#{old_val ? 'Yes' : 'No'}\" to \"#{new_val ? 'Yes' : 'No'}\")" if has_changed
-      when 'FloatSymptom', 'IntegerSymptom'
-        delta << "#{symptom.label} (\"#{old_val}\" to \"#{new_val}\")" if new_val != old_val
+    Assessment.transaction do
+      assessment.reported_condition&.symptoms&.each do |symptom|
+        next unless reported_symptoms.key?(symptom[:name])
+
+        new_val = reported_symptoms[symptom[:name]][:value]
+        old_val = symptom.value
+        next if new_val&.to_s == old_val&.to_s
```

Comment:
I think doing this .to_s might be the cause of this issue where float values are marked as different in the history item even if no changes are made 
![image](https://user-images.githubusercontent.com/3967953/116732106-a4451700-a9b8-11eb-85da-84f236bc1b8d.png)

Reply:
I remember changing this because of a failing test case but I don't remember which one, so I pushed up this change and am going to wait to see which one fails

Reply:
seems like that's not the case and the tests have passed, here is the change: https://github.com/SaraAlert/SaraAlert/pull/852/commits/58dc67c0b1988c047836deb096c3f03ee45b0a6a